docker-images/jenkins/Dockerfile

```diff
@@ -12,41 +12,44 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-FROM jenkins/jenkins:lts
-USER root
+FROM jenkins/jenkins:2.190.3
+LABEL maintainer="WSO2 Docker Maintainers <dev@wso2.org>"
 
-#Pre-Install Jenkins Plugins
-COPY plugins.txt /usr/share/jenkins/ref/plugins.txt
-RUN /usr/local/bin/install-plugins.sh < /usr/share/jenkins/ref/plugins.txt
+USER root
 
 # Install Docker
-RUN apt-get update && apt-get install software-properties-common apt-transport-https ca-certificates -y; \
-curl -fsSL https://download.docker.com/linux/debian/gpg | apt-key add -;\
-add-apt-repository  "deb [arch=amd64] https://download.docker.com/linux/debian  $(lsb_release -cs) stable";\
-apt-get update && apt-get install docker-ce -y
-
-# Install maven
-RUN apt-get install maven -y
+RUN \
+    apt-get update \
+    && apt-get install -y \
+        software-properties-common \ 
+        apt-transport-https \
+        ca-certificates \
+    && curl -fsSL https://download.docker.com/linux/debian/gpg | apt-key add - \
+    && add-apt-repository  "deb [arch=amd64] https://download.docker.com/linux/debian  $(lsb_release -cs) stable" \
+    && apt-get update \
+    && apt-get install docker-ce -y
```

Comment:
@Aaquiff let us combine all the the packages installed via APT under a single command. Also, ensure that the packages are in the alphabetical order in terms of their names' starting letter (recommended best practice [1]).

Move the addition of APT repository to the top and run a single APT package list update.

Also, please clean up the apt cache by removing `/var/lib/apt/lists` in order to reduce the image size, further. Please see [2] for advanced guidelines on managing packages.

[1]: [Sort multi-line arguments](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#sort-multi-line-arguments)
[2]: [APT-GET](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#run)

Reply:
@chirangaalwis add-apt-repository is not available by default. Therefore we need to install **software-properties-common** package before adding a repository.

I will do the reordering of the packages and cache cleanup changes.